tests/test_meshtastic_utils_connect.py

```
@@ -626,3 +825,1141 @@ def test_none_startup_drain_event_is_safe_noop(reset_meshtastic_globals):
+        assert mu.meshtastic_client is None
+        assert mu._relay_active_client_id is None
+
+    def test_cleanup_early_return_when_client_changed(self):
+        mock_client = MagicMock()
+
+        other_client = MagicMock()
+
+        def _change_client_side_effect() -> dict[str, dict[str, str]]:
+            mu.meshtastic_client = other_client
+            mu._relay_active_client_id = id(other_client)
+            raise RuntimeError("trigger cleanup")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)